lib/apiservers/engine/backends/container.go

```diff
@@ -1605,6 +1599,70 @@ func (c *Container) containerAttach(name string, ca *backend.ContainerAttachConf
 	return nil
 }
 
+// ContainerRename changes the name of a container, using the oldName
+// to find the container. An error is returned if newName is already
+// reserved.
+func (c *Container) ContainerRename(oldName, newName string) error {
+	defer trace.End(trace.Begin(newName))
+
+	if oldName == "" || newName == "" {
+		err := fmt.Errorf("neither old nor new names may be empty")
+		log.Errorf("%s", err.Error())
+		return err
+	}
+
+	if !utils.RestrictedNamePattern.MatchString(newName) {
+		err := fmt.Errorf("invalid container name (%s), only %s are allowed", newName, utils.RestrictedNameChars)
+		log.Errorf("%s", err.Error())
```

Comment:
This is not an error. It's just warning. I also would change the error message to:

"Container name (%s) contains not allowed characters. Only %s characters are allowed."

Since this message sent to the client, phrasing does matter.